code/modules/tools/tool_archetype_definitions_welder.dm

```diff
@@ -0,0 +1,28 @@
+/decl/tool_archetype/welder
+	name = "welder"
+	use_sound = list('sound/items/Welder.ogg','sound/items/Welder2.ogg')
+	codex_key = TOOL_CODEX_WELDER
+
+/decl/tool_archetype/welder/handle_pre_interaction(var/mob/user, var/atom/tool, var/expend_fuel = 0)
```

Comment:
I feel like you also have to do the can_use check you do in the main interaction proc here, unless you do it in whatever calls this.

Reply:
Yep, I got confused a bit and never actually called can_use_tool in main code.